google/cloud/spanner/date.h

```diff
@@ -40,6 +41,9 @@ class Date {
   int month() const { return month_; }
   int day() const { return day_; }
 
+  /// @name Output streaming
+  friend std::ostream& operator<<(std::ostream& os, Date const& date);
```

Comment:
Why does it need to be a friend?

Reply:
It doesn't technically need to be a friend. But the reasons why I thought it made the most sense declare it as an inline friend like this are:

1. All other `op<<` declarations in our code are inline friends, so making this one the same seems nice.

2. There are other (minor) benefits to inline friend declarations, such as they can *only* be found via ADL (and therefore work well with operators), they don't add to overloads in the namespace thus making compile-times faster, etc.

But if you'd prefer this be an out-of-line non-friend, I'll change it.

Reply:
Nevermind. I removed the friendship.

Reply:
And if you'd prefer to keep it, that's fine too.  But the relational operators should probably match.